### PR TITLE
Reestablish connection when mixclient fails

### PR DESCRIPTION
**Pull request by al8n**
In the current implementation, if `nomos-1` node is connected with `mixnode-1` , and if `mixnode-1` becomes down and is restarted quickly, the `nomos-1` does not re-establish the connection with `mixnode-1`.

This PR introduces a loop, so when the mixnet client fails, we reestablish the connection between `nomos-1` and `mixnet-1`.

### CODE REVIEW

File: nomos-services/network/src/backends/libp2p/mixnet.rs [hunk 1]
```diff
@@ -47,11 +47,22 @@ impl MixnetHandler {
...
 
     pub async fn run(&mut self) {
-        let Ok(mut stream) = self.client.run().await else {
-            tracing::error!("Could not quickstart mixnet stream");
-            return;
-        };
+        // we need this loop to help us reestablish the connection in case
+        // the mixnet client fails for whatever reason
+        loop {
+            tokio::select! {
```
Comment (zeegomo) [inline nomos-services/network/src/backends/libp2p/mixnet.rs:53]: why as select if there's only one branch?

File: nomos-services/network/src/backends/libp2p/mixnet.rs [hunk 2]
```diff
@@ -47,11 +47,22 @@ impl MixnetHandler {
...
+                stream = self.client.run() => {
+                    if let Ok(stream) = stream {
+                        Self::handle_stream(self.commands_tx.clone(), stream).await;
+                    } else {
+                        tracing::error!("Could not quickstart mixnet stream");
```
Comment (zeegomo) [inline nomos-services/network/src/backends/libp2p/mixnet.rs:58]: we should add a bit of wait in between retries

File: nomos-services/network/src/backends/libp2p/mixnet.rs (hunk 1 above)
Comment (al8n) [reply nomos-services/network/src/backends/libp2p/mixnet.rs]: Ah, it is a left-over.

**Review by youngjoon-lee — APPROVED** [review]

**Review by zeegomo — APPROVED** [review]

**Comment by youngjoon-lee** [comment]
@al8n Do you have any idea why CI fails with this error?
https://github.com/logos-co/nomos-node/actions/runs/6558398269/job/17854307256?pr=445
If this error is related to this PR, we should fix it. But, if not, we need to merge this PR to unblock #467 
```
---- ten_nodes_happy stdout ----
waiting... 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0
waiting... 0 | 1 | 1 | 1 | 1 | 1 | 1 | 1 | 1 | 1
waiting... 1 | 1 | 1 | 1 | 1 | 1 | 1 | 1 | 1 | 1
waiting... 2 | 2 | 2 | 2 | 2 | 2 | 2 | 2 | 2 | 2
waiting... 3 | 3 | 3 | 3 | 3 | 3 | 3 | 3 | 3 | 3
waiting... 3 | 3 | 3 | 3 | 3 | 3 | 3 | 3 | 3 | 3
waiting... 4 | 4 | 4 | 4 | 4 | 4 | 4 | 4 | 4 | 4
waiting... 5 | 5 | 5 | 5 | 5 | 5 | 5 | 5 | 5 | 5
waiting... 5 | 5 | 5 | 5 | 5 | 5 | 5 | 5 | 5 | 5
waiting... 6 | 6 | 6 | 6 | 6 | 6 | 6 | 6 | 6 | 6
waiting... 6 | 6 | 6 | 6 | 6 | 6 | 6 | 6 | 6 | 6
waiting... 7 | 7 | 7 | 7 | 7 | 7 | 7 | 7 | 7 | 7
waiting... 7 | 7 | 7 | 7 | 7 | 7 | 7 | 7 | 7 | 7
waiting... 7 | 7 | 7 | 7 | 7 | 7 | 7 | 7 | 7 | 7
waiting... 8 | 8 | 8 | 8 | 8 | 8 | 8 | 8 | 8 | 8
waiting... 8 | 8 | 8 | 8 | 8 | 8 | 8 | 8 | 8 | 8
waiting... 9 | 9 | 9 | 9 | 9 | 9 | 9 | 9 | 9 | 9
waiting... 9 | 9 | 9 | 9 | 9 | 9 | 9 | 9 | 9 | 10
waiting... 10 | 10 | 10 | 10 | 10 | 10 | 10 | 10 | 10 | 10
waiting... 10 | 10 | 10 | 10 | 10 | 10 | 10 | 10 | 10 | 10
waiting... 11 | 11 | 11 | 11 | 11 | 11 | 11 | 11 | 11 | 11
waiting... 11 | 11 | 11 | 11 | 11 | 11 | 11 | 11 | 11 | 11
waiting... 12 | 12 | 12 | 12 | 12 | 12 | 12 | 12 | 12 | 12
waiting... 12 | 12 | 12 | 12 | 12 | 12 | 12 | 12 | 12 | 12
waiting... 13 | 13 | 13 | 13 | 13 | 13 | 13 | 13 | 13 | 13
waiting... 13 | 13 | 13 | 13 | 13 | 13 | 13 | 13 | 13 | 13
waiting... 14 | 14 | 14 | 14 | 14 | 14 | 14 | 14 | 14 | 14
waiting... 14 | 14 | 14 | 14 | 15 | 15 | 15 | 15 | 15 | 15
waiting... 15 | 15 | 15 | 15 | 15 | 15 | 15 | 15 | 15 | 15
waiting... 16 | 16 | 16 | 16 | 16 | 16 | 16 | 16 | 16 | 16
waiting... 16 | 16 | 16 | 16 | 16 | 16 | 16 | 16 | 16 | 16
waiting... 17 | 17 | 17 | 17 | 17 | 17 | 17 | 17 | 17 | 17
waiting... 17 | 17 | 17 | 17 | 17 | 17 | 17 | 17 | 17 | 17
thread 'ten_nodes_happy' panicked at tests/src/tests/happy.rs:21:24:
timed out waiting for nodes to reach view 20
```

**Comment by al8n** [comment]
I have no idea about these CI failures, I do not think this related to this PR.

**Pull request merged**